test/functional/ui/statusline_spec.lua

```diff
@@ -1,7 +1,72 @@
 local helpers = require('test.functional.helpers')(after_each)
 local Screen = require('test.functional.ui.screen')
-local clear, command, feed = helpers.clear, helpers.command, helpers.feed
-local eq, funcs, meths = helpers.eq, helpers.funcs, helpers.meths
+local clear = helpers.clear
+local command = helpers.command
+local feed = helpers.feed
+local eq = helpers.eq
+local funcs = helpers.funcs
+local meths = helpers.meths
+local exec = helpers.exec
+
+local function lastmessage()
+  return funcs.execute('1messages'):sub(2)
+end
+
+describe('statusline clicks', function()
+  local screen
+
+  before_each(function()
+    clear()
+    screen = Screen.new(40, 8)
+    screen:attach()
+    command('set laststatus=2')
+    exec([=[
+      function! MyClickFunc(minwid, clicks, button, mods)
+        call nvim_echo([[ printf("%d %d %s", a:minwid, a:clicks, a:button) ]], v:true, {})
```

Comment:
Will setting a global Vim string variable (`let g:`) and getting or evaluating it be better?

Reply:
I think it would, so I changed it to do that